webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingResultProcessor.java

```diff
@@ -0,0 +1,105 @@
+package com.box.l10n.mojito.service.thirdparty.smartling;
+
+import com.amazonaws.SdkClientException;
+import com.box.l10n.mojito.service.blobstorage.Retention;
+import com.box.l10n.mojito.service.blobstorage.s3.S3BlobStorage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@ConditionalOnProperty(value = "l10n.ThirdPartyTMS.impl", havingValue = "ThirdPartyTMSSmartling")
+@Component
+public class SmartlingResultProcessor {
+
+    static Logger logger = LoggerFactory.getLogger(SmartlingResultProcessor.class);
+
+    private static final String NAME_FORMAT = "%s_%s_%s.zip";
+
+    @Autowired
+    S3BlobStorage s3BlobStorage;
+
+    public String processPush(List<SmartlingFile> files,
```

Comment:
probably won't need to return s3url in those functions when using `StructuredBlobStorage`

Reply:
This is specifically meant to be uploaded to S3, not through the generic interface.

Reply:
ah to make the download easy? 

So I guess the dry run option won't be useful unless using S3. Probably fine since this is mostly for testing I guess.

Reply:
Ultimately that's the goal, to use S3 as the storage for testing purposes and nothing else.

Reply:
ok, i was thinking about people wanting to use that without s3 but yeah not needed at this point since it is mostly to test the implementation